src/vhost_kern/mod.rs

```diff
@@ -49,15 +48,21 @@ pub trait VhostKernBackend<'a>: AsRawFd {
     /// Check whether the ring configuration is valid.
     fn is_valid(&self, config_data: &VringConfigData) -> bool {
         let queue_size = config_data.queue_size;
-        let desc_table_size = 16 * u64::from(queue_size) as GuestUsize;
-        let avail_ring_size = 6 + 2 * u64::from(queue_size) as GuestUsize;
-        let used_ring_size = 6 + 8 * u64::from(queue_size) as GuestUsize;
         if queue_size > config_data.queue_max_size
             || queue_size == 0
             || (queue_size & (queue_size - 1)) != 0
         {
-            false
-        } else if GuestAddress(config_data.desc_table_addr)
+            return false;
+        }
+
+        // TODO: the GuestMemory trait lacks of method to look up GPA by HVA,
```

Comment:
What would this lookup method look like? Either the HVA is mapped into the guest physical address space and gpa=hva, or else it'd return an error?

Reply:
I think it would be something like :
```
fn get_gpa_by_hva(&self, hva: u64) -> Result<GuestAddress>
```